core/src/main/java/org/mule/runtime/core/api/config/DefaultMuleConfiguration.java

```diff
@@ -476,6 +485,14 @@ public void setEnableStreaming(boolean enableStreaming) {
     }
   }
 
+  private boolean isLazyInit() {
```

Comment:
Will this method be called multiple times? Can the config value change?
Wouldn't it be better to use a LazyValue to avoid resolving the configuration property everytime?

Reply:
ConfigurationProperties already has a cache to avoid resolving through the whole hierarchy the same key over and over.

Reply:
I couldn't find where the cache is so I ended caching the lazyInit flag here.